jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/DijkstraShortestPath.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.shortestpath;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+
+/**
+ * An implementation of <a href="http://mathworld.wolfram.com/DijkstrasAlgorithm.html">Dijkstra's
+ * shortest path algorithm</a> using a Fibonacci heap.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author John V. Sichi
+ */
+public final class DijkstraShortestPath<V, E>
+    extends BaseShortestPathAlgorithm<V, E>
+{
+    private final double radius;
+
+    /**
+     * Constructs a new instance of the algorithm for a given graph.
+     * 
+     * @param graph the graph
+     */
+    public DijkstraShortestPath(Graph<V, E> graph)
+    {
+        this(graph, Double.POSITIVE_INFINITY);
+    }
+
+    /**
+     * Constructs a new instance of the algorithm for a given graph.
+     *
+     * @param graph the graph
+     * @param radius limit on path length, or Double.POSITIVE_INFINITY for unbounded search
+     */
+    public DijkstraShortestPath(Graph<V, E> graph, double radius)
+    {
+        super(graph);
+        if (radius < 0.0) {
+            throw new IllegalArgumentException("Radius must be non-negative");
+        }
+        this.radius = radius;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public GraphPath<V, E> getPath(V source, V sink)
+    {
+        if (!graph.containsVertex(source)) {
+            throw new IllegalArgumentException("graph must contain the source vertex");
+        }
+        if (!graph.containsVertex(sink)) {
+            throw new IllegalArgumentException("graph must contain the sink vertex");
+        }
+        if (source.equals(sink)) {
+            return createEmptyPath(source, sink);
+        }
+
+        DijkstraClosestFirstIterator<V, E> it =
+            new DijkstraClosestFirstIterator<>(graph, source, radius);
+
+        while (it.hasNext()) {
+            V vertex = it.next();
+            if (vertex.equals(sink)) {
+                break;
+            }
+        }
+
+        return it.getPaths().getPath(sink);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SingleSourcePaths<V, E> getPaths(V source)
+    {
+        if (!graph.containsVertex(source)) {
+            throw new IllegalArgumentException("graph must contain the source vertex");
+        }
+
+        DijkstraClosestFirstIterator<V, E> it =
+            new DijkstraClosestFirstIterator<>(graph, source, radius);
+
+        while (it.hasNext()) {
+            it.next();
+        }
+
+        return it.getPaths();
+    }
+
```

Comment:
I reintroduced the method in each class. It is difficult (naming-wise) to have a package-level utility class since different algorithms make different assumptions.